src/components/Footer.tsx

```diff
@@ -0,0 +1,117 @@
+import classNames from 'classnames';
+import { useEffect } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { Todo } from '../types/Todo';
+import { getSearchWith } from '../utils/getSearchWith';
+
+type Props = {
+  visibleTodos: Todo[],
+  setVisibleTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+  todos: Todo[],
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+};
+
+enum FilterType {
+  All = 'all',
+  Completed = 'completed',
+  Active = 'active',
+}
+
+export const Footer = ({
+  todos,
+  setTodos,
+  setVisibleTodos,
+  visibleTodos,
+}: Props) => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const filterType = searchParams.get('filter') || FilterType.All;
```

Comment:
Wrap the filterType value in useMemo, please. It's necessary because you use it in the useEffect dependencies